homeassistant/components/reolink/light.py

```diff
@@ -0,0 +1,157 @@
+"""Component providing support for Reolink light entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import Host
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ColorMode,
+    LightEntity,
+    LightEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkLightEntityDescriptionMixin:
+    """Mixin values for Reolink light entities."""
+
+    is_on_fn: Callable[[Host, int], bool]
+    turn_on_off_fn: Callable[[Host, int, bool], Any]
+
+
+@dataclass
+class ReolinkLightEntityDescription(
+    LightEntityDescription, ReolinkLightEntityDescriptionMixin
+):
+    """A class that describes light entities."""
+
+    supported_fn: Callable[[Host, int], bool] = lambda api, ch: True
+    get_brightness_fn: Callable[[Host, int], int] | None = None
+    set_brightness_fn: Callable[[Host, int, float], Any] | None = None
+
+
+LIGHT_ENTITIES = (
+    ReolinkLightEntityDescription(
+        key="floodlight",
+        name="Floodlight",
+        icon="mdi:spotlight-beam",
+        supported_fn=lambda api, ch: api.supported(ch, "floodLight"),
+        is_on_fn=lambda api, ch: api.whiteled_state(ch),
+        turn_on_off_fn=lambda api, ch, value: api.set_whiteled(ch, state=value),
+        get_brightness_fn=lambda api, ch: api.whiteled_brightness(ch),
+        set_brightness_fn=lambda api, ch, value: api.set_whiteled(
+            ch, brightness=int(value)
```

Comment:
brightness is already an int, by the validation schema.

Reply:
Thanks, I have corrected this